src/proto/graplinc/grapl/api/services/v1beta1/types.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package graplinc.grapl.api.services.v1beta1;
+
+// A protobuf uuid, encoded by splitting the 128bit representation into two 64bit values
+message ProtoUuidV4 {
+  // Least significant bytes
+  fixed64 lsb = 1;
+  // Most significant bytes
+  fixed64 msb = 2;
+}
+
+// Metadata about a specific message, or its context
+message Meta {
+  // The name of the service/ client that produced this message
+  string client_id = 1;
+  // Customer identity
+  ProtoUuidV4 tenant_id = 2;
+  // Unique identity representing a unified request throughout Grapl
+  ProtoUuidV4 trace_id = 3;
+}
+
+// A generic wrapper around other messages
+message Envelope {
+  // Metadata about this message
+  Meta metadata = 1;
+  // The string name of this message's type
```

Comment:
More specifically, this needs to be the fully-qualified name of the protobuf type used to serialize this message. This will become important later because upon deserialization we will need to look up the appropriate deserializer.